gamemodes/terrortown/gamemode/client/cl_main.lua

```diff
@@ -311,8 +311,29 @@ local function PlaySoundCue()
 	surface.PlaySound(cues[math.random(#cues)])
 end
 
-GM.TTTBeginRound = PlaySoundCue
-GM.TTTEndRound = PlaySoundCue
+---
+-- A hook that is called when the preparation phase starts
+-- @hook
+-- @realm client
+function GM:TTTPrepareRound()
+	EPOP:Clear()
```

Comment:
I think we should discuss, if this would not rather be placed in https://github.com/TTT-2/TTT2/blob/3008bec9301eff74bf07302736733d0cda34266d/gamemodes/terrortown/gamemode/client/cl_main.lua#L281

or if it even makes a difference, eg. is this due to being in a hook, also not deterministic and should this be the way to add code, when a hook usually is used to let external addons be called instead of ours? And why did TTT do it like that, with the `PlaySoundCue`. This seems inconsistent to me and I'd like to hear @Alf21's opinion in this.

Reply:
> I think we should discuss, if this would not rather be placed in https://github.com/TTT-2/TTT2/blob/3008bec9301eff74bf07302736733d0cda34266d/gamemodes/terrortown/gamemode/client/cl_main.lua#L281
> 
> or if it even makes a difference, eg. is this due to being in a hook, also not deterministic and should this be the way to add code, when a hook usually is used to let external addons be called instead of ours? And why did TTT do it like that, with the `PlaySoundCue`. This seems inconsistent to me and I'd like to hear @Alf21's opinion in this.

Hm. I don't see why this isn't deterministic tbh. And there are many hooks used like this in TTT2.